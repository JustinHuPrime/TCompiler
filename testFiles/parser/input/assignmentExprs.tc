module foo;

void bar(int i, int j) {
  i = j *= i /= j %= i += j -= i <<= j >>= i >>>= j &= i ^= j |= i;
  bool b = true;
  b &&= false ||= true;
}