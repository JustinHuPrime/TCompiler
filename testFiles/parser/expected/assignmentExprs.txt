testFiles/parser/assignmentExprs.tc (code):
FILE(1, 1, STAB(ENTRY(bar, FUNCTION(testFiles/parser/assignmentExprs.tc, 3, 1, void(int, int)))), MODULE(1, 1, ID(1, 8, foo, REFERENCES())), FUNDEFN(3, 1, KEYWORDTYPE(3, 1, void), ID(3, 6, bar, REFERENCES(testFiles/parser/assignmentExprs.tc, 3, 1)), KEYWORDTYPE(3, 10, int), KEYWORDTYPE(3, 17, int), ID(3, 14, i, REFERENCES()), ID(3, 21, j, REFERENCES()), STAB(ENTRY(i, VARIABLE(testFiles/parser/assignmentExprs.tc, 3, 10, int)), ENTRY(j, VARIABLE(testFiles/parser/assignmentExprs.tc, 3, 17, int))), COMPOUNDSTMT(3, 24, STAB(ENTRY(b, VARIABLE(testFiles/parser/assignmentExprs.tc, 5, 8, bool))), EXPRESSIONSTMT(4, 3, BINOPEXP(4, 3, ASSIGN, ID(4, 3, i, REFERENCES(testFiles/parser/assignmentExprs.tc, 3, 10)), BINOPEXP(4, 7, MULASSIGN, ID(4, 7, j, REFERENCES(testFiles/parser/assignmentExprs.tc, 3, 17)), BINOPEXP(4, 12, DIVASSIGN, ID(4, 12, i, REFERENCES(testFiles/parser/assignmentExprs.tc, 3, 10)), BINOPEXP(4, 17, MODASSIGN, ID(4, 17, j, REFERENCES(testFiles/parser/assignmentExprs.tc, 3, 17)), BINOPEXP(4, 22, ADDASSIGN, ID(4, 22, i, REFERENCES(testFiles/parser/assignmentExprs.tc, 3, 10)), BINOPEXP(4, 27, SUBASSIGN, ID(4, 27, j, REFERENCES(testFiles/parser/assignmentExprs.tc, 3, 17)), BINOPEXP(4, 32, LSHIFTASSIGN, ID(4, 32, i, REFERENCES(testFiles/parser/assignmentExprs.tc, 3, 10)), BINOPEXP(4, 38, ARSHIFTASSIGN, ID(4, 38, j, REFERENCES(testFiles/parser/assignmentExprs.tc, 3, 17)), BINOPEXP(4, 44, LRSHIFTASSIGN, ID(4, 44, i, REFERENCES(testFiles/parser/assignmentExprs.tc, 3, 10)), BINOPEXP(4, 51, BITANDASSIGN, ID(4, 51, j, REFERENCES(testFiles/parser/assignmentExprs.tc, 3, 17)), BINOPEXP(4, 56, BITXORASSIGN, ID(4, 56, i, REFERENCES(testFiles/parser/assignmentExprs.tc, 3, 10)), BINOPEXP(4, 61, BITORASSIGN, ID(4, 61, j, REFERENCES(testFiles/parser/assignmentExprs.tc, 3, 17)), ID(4, 66, i, REFERENCES(testFiles/parser/assignmentExprs.tc, 3, 10))))))))))))))), VARDEFNSTMT(5, 3, KEYWORDTYPE(5, 3, bool), ID(5, 8, b, REFERENCES(testFiles/parser/assignmentExprs.tc, 5, 8)), LITERAL(5, 12, BOOL(true))), EXPRESSIONSTMT(6, 3, BINOPEXP(6, 3, LANDASSIGN, ID(6, 3, b, REFERENCES(testFiles/parser/assignmentExprs.tc, 5, 8)), BINOPEXP(6, 9, LORASSIGN, LITERAL(6, 9, BOOL(false)), LITERAL(6, 19, BOOL(true))))))))
